src/Command/RaiseToInstalledCommand.php

```diff
@@ -0,0 +1,164 @@
+            }
+
+            $installedVersion = $installedPackagesToVersions[$packageName];
+
+            // special case for unions
+            if (str_contains((string) $packageVersion, '|')) {
+                $passingVersionKeys = [];
+
+                $unionPackageVersions = explode('|', (string) $packageVersion);
+                foreach ($unionPackageVersions as $key => $unionPackageVersion) {
+                    $unionPackageConstraint = $this->versionParser->parseConstraints($unionPackageVersion);
+
+                    if (Comparator::greaterThanOrEqualTo(
+                        $installedVersion,
+                        $unionPackageConstraint->getLowerBound()
+                            ->getVersion()
+                    )) {
+                        $passingVersionKeys[] = $key;
+                    }
+                }
+
+                // nothing we can do, as lower union version is passing
+                if ($passingVersionKeys === [0]) {
+                    continue;
+                }
+
+                // higher version is meet, let's drop the lower one
+                if ($passingVersionKeys === [0, 1]) {
+                    $newPackageVersion = $unionPackageVersions[1];
+
+                    $composerJsonContents = ComposerJsonPackageVersionUpdater::update(
+                        $composerJsonContents,
+                        $packageName,
+                        $newPackageVersion
+                    );
+
+                    $hasChanged = true;
+                    continue;
+                }
+            }
+
+            $normalizedInstalledVersion = $this->versionParser->normalize($installedVersion);
+            $installedPackageConstraint = $this->versionParser->parseConstraints($packageVersion);
+
+            $normalizedConstraintVersion = $this->versionParser->normalize(
+                $installedPackageConstraint->getLowerBound()
+                    ->getVersion()
+            );
+
+            // remove "-dev" suffix
+            $normalizedConstraintVersion = str_replace('-dev', '', $normalizedConstraintVersion);
+
+            // all equal
+            if ($normalizedConstraintVersion === $normalizedInstalledVersion) {
+                continue;
+            }
+
+            [$major, $minor, $patch] = explode('.', $normalizedInstalledVersion);
+
+            $newRequiredVersion = sprintf('^%s.%s', $major, $minor);
+
+            // lets update
+            $composerJsonContents = ComposerJsonPackageVersionUpdater::update(
+                $composerJsonContents,
+                $packageName,
+                $newRequiredVersion
+            );
+
+            $hasChanged = true;
+            continue;
+            // focus on minor only
+            // or on patch in case of 0.*
+        }
+
+        if ($hasChanged) {
+            $symfonyStyle->success('Updating "composer.json" with installed versions');
+            FileSystem::write($composerJsonFilePath, $composerJsonContents);
```

Comment:
ensure pass `null` argument to 3rd parameter to ensure the file mode is not changed:

```suggestion
            FileSystem::write($composerJsonFilePath, $composerJsonContents, null);
```

see why https://github.com/nette/utils/blob/2d3e496c5cb2a7189e3f746c644fceb4f5f88add/src/Utils/FileSystem.php#L223 , otherwise, it forced changed to `666`

Reply:
Got it, thanks